core/views.py

```diff
@@ -51,3 +54,89 @@ class DocumentUpdateView(SuccessMessageMixin, UpdateView):
 
     def get_success_url(self):
         return self.request.POST.get("next")
+
+
+class ContactAddFormView(FormView):
+    template_name = "core/_contact_add_form.html"
+    form_class = ContactAddForm
+
+    def get_initial(self):
+        initial = super().get_initial()
+        initial["fiche_id"] = self.request.GET.get("fiche_id")
+        initial["next"] = self.request.GET.get("next")
+        initial["content_type_id"] = self.request.GET.get("content_type_id")
+        return initial
+
+    def form_valid(self, form):
+        selection_form = ContactSelectionForm(
+            structure=form.cleaned_data.get("structure"),
+            fiche_id=form.cleaned_data.get("fiche_id"),
+            content_type_id=form.cleaned_data.get("content_type_id"),
+            initial={
+                "next": form.cleaned_data.get("next"),
+            },
+        )
+        return render(self.request, self.template_name, {"form": form, "selection_form": selection_form})
+
+
+class ContactSelectionView(FormView):
+    template_name = "core/_contact_add_form.html"
+    form_class = ContactSelectionForm
+
+    # todo ajouter next dans context pour lien de retour vers la fiche
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        print(self.request.POST)
+        context["next"] = self.request.GET.get("next")
+        return context
+
+    def get_form_kwargs(self):
+        kwargs = super(ContactSelectionView, self).get_form_kwargs()
+        kwargs.update(
+            {
+                "structure": self.request.POST.get("structure", ""),
+                "fiche_id": self.request.POST.get("fiche_id", ""),
+                "content_type_id": self.request.POST.get("content_type_id", ""),
+            }
+        )
+        return kwargs
+
+    def form_valid(self, form):
+        content_type = ContentType.objects.get(pk=form.cleaned_data["content_type_id"]).model_class()
+        fiche = content_type.objects.get(pk=form.cleaned_data["fiche_id"])
+        contacts = form.cleaned_data["contacts"]
+        for contact in contacts:
+            fiche.contacts.add(contact)
+        fiche.save()
```

Comment:
De mémoire : pas besoin de save après une opération comme .add ou .set sur un many to many.